public/app/features/alerting/unified/plugins/PluginOriginBadge.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { useAsync } from 'react-use';
+
+import { Badge, Tooltip } from '@grafana/ui';
+
+import { getPluginSettings } from '../../../plugins/pluginSettings';
+
+interface PluginOriginBadgeProps {
+  pluginId: string;
+}
+
+export function PluginOriginBadge({ pluginId }: PluginOriginBadgeProps) {
+  const { value: pluginMeta } = useAsync(() => getPluginSettings(pluginId));
```

Comment:
Don't we want to use the `loading` state here and return `null`? (Or a loading spinner)

Just to make it clearer about the async nature of this badge?

Reply:
Tough question. Currently `<Badge text={pluginId} color="orange" />` works as placeholder for missing `logo` information that also includes the loading state
For most cases plugins' settings will be in cache so no actual loading should happen
I don't know if rendering `null` is better than a badge with a `pluginId` 🤔

Reply:
I guess it's fine then for it to be as is :+1: